### PR TITLE
TRUS-2369 Remove class of beneficiary

### PR DESCRIPTION
**Pull request by JoeSimmonds**

### CODE REVIEW

File: app/uk/gov/hmrc/trusts/services/BeneficiaryTransformationService.scala [hunk 1]
```diff
@@ -17,19 +17,47 @@
...
+import uk.gov.hmrc.trusts.models.{RemoveBeneficiary, Success}
 import uk.gov.hmrc.trusts.transformers._
 
-import scala.concurrent.Future
+import scala.concurrent.{ExecutionContext, Future}
 
-class BeneficiaryTransformationService @Inject()(transformationService: TransformationService) {
+class BeneficiaryTransformationService @Inject()(transformationService: TransformationService)(implicit ec:ExecutionContext) {
+  def removeBeneficiary(utr: String, internalId: String, removeBeneficiary: RemoveBeneficiary)(implicit hc: HeaderCarrier) : Future[Success.type] = {
+    transformationService.getTransformedData(utr, internalId).flatMap {
```
Comment (adamconder) [inline app/uk/gov/hmrc/trusts/services/BeneficiaryTransformationService.scala]: @JoeSimmonds Could we extract the core of this 'picking' into a separate function named 'getBeneficiaryJson(type, index)' which returns a the future we can map over?

File: app/uk/gov/hmrc/trusts/transformers/RemoveBeneficiariesTransform.scala [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        )
+    }
+
+    val xform = path.json.update(removeFromArray)
+
+    input.transform(xform)
+  }
+
+  override def applyDeclarationTransform(input: JsValue): JsResult[JsValue] = {
+    super.applyDeclarationTransform(input)
```
Comment (adamconder) [inline app/uk/gov/hmrc/trusts/transformers/RemoveBeneficiariesTransform.scala]: Do we need to call super.applyDeclarationTransform here?

Comment (JoeSimmonds) [reply app/uk/gov/hmrc/trusts/transformers/RemoveBeneficiariesTransform.scala]: probably not

File: app/uk/gov/hmrc/trusts/services/BeneficiaryTransformationService.scala (hunk 1 above)
Comment (JoeSimmonds) [reply app/uk/gov/hmrc/trusts/services/BeneficiaryTransformationService.scala]: Not sure if this is what you meant

**Pull request merged**